src/editor/Core/Utility.re

```diff
@@ -37,8 +37,9 @@ let getFileContents = (path, ~handler) => {
 let convertUTF8string = str =>
   CamomileLibraryDefault.Camomile.(UChar.code(UTF8.get(str, 0)));
 
-let convertNeovimExtType = (buffer: Msgpck.t) =>
-  switch (buffer) {
-  | Msgpck.Ext(kind, id) => Some((kind, convertUTF8string(id)))
-  | _ => None
+let safe_fold_left2 = (fn, accum, list1, list2, ~default) =>
+  try (List.fold_left2(fn, accum, list1, list2)) {
+  | Invalid_argument(reason) =>
```

Comment:
Interesting - is this an expected scenario? When does this fail - different size lists?

Reply:
@bryphe it fails in development when I'm messing around its not an expected failure/ doesn't occur when I'm not changing the inputs but it takes down the app which is annoying for development :laughing:

Reply:
Sorry realise that wasn't the specific answer yes different size lists will cause it to fail but technically that shouldn't be possible without messing with the inputs since one request is based on the size of another e.g. I ask for all the buffers get a list of length 3 then I ask for metadata on the buffers -> get a result of list 3 (metadata items)

The way it breaks in development is when I've tweaked this sometimes I get the calls wrong and the app crashes till I fix the broken call but it'd be easier to debug if it returned a default and logged the error